server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -426,8 +426,9 @@ protected void updateCompactionCompleted(TExternalCompactionJob job, TCompaction
    * @throws RetriesExceededException thrown when retries have been exceeded
    */
   protected TExternalCompactionJob getNextJob(Supplier<UUID> uuid) throws RetriesExceededException {
+    final long checkWaitTime = getConfiguration().getTimeInMillis(Property.COMPACTOR_JOB_WAIT_TIME);
     RetryableThriftCall<TExternalCompactionJob> nextJobThriftCall =
-        new RetryableThriftCall<>(1000, RetryableThriftCall.MAX_WAIT_TIME, 0, () -> {
+        new RetryableThriftCall<>(checkWaitTime, checkWaitTime, 0, () -> {
```

Comment:
With this change there is no longer exponential backoff when the queue keeps being empty.  The following would still allow the exponential backoff.

```suggestion
        long maxWaitTime = getConfiguration().getTimeInMillis(Property.COMPACTOR_JOB_WAIT_MAX_TIME);
        long minWaitTime = Math.min(1000, maxWaitTime);
        new RetryableThriftCall<>(minWaitTime, maxWaitTime, 0, () -> {
```